### PR TITLE
[Self-service error codes] Use error validators in ApiPackageManagementService

### PR DESCRIPTION
**Pull request by tudor-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compatibility/bazel_tools/testing.bzl [hunk 1]
```diff
@@ -546,6 +546,18 @@ excluded_test_tool_tests = [
...
+    {
+        # Self-service error code assertions adapted
+        "start": "1.17.1",
+        "platform_ranges": [
+            {
+                "end": "1.18.0-snapshot.20211102.8257.0.7391a3cd",
+                "exclusions": [
+                    "PackageManagementServiceIT",
+                ],
+            },
```
Comment (pbatko-da) [inline compatibility/bazel_tools/testing.bzl:558]: Does this mean: "do not run PackageManagementServiceIT if ledger-api-test-tool is 1.17.1 or newer, against ledger server version 1.18.0-snapshot.20211102.8257.0.7391a3cd or older"?

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPackageManagementService.scala [hunk 2]
```diff
@@ -94,56 +98,60 @@ private[apiserver] final class ApiPackageManagementService private (
...
         }
-
         response.andThen(logger.logErrorsOnCall[UploadDarFileResponse])
     }
+
+  private implicit class ErrorValidations[E, R](result: Either[E, R]) {
+    def loggingValidation(validate: E => BaseError.Impl): Try[R] =
+      result.left.map { err =>
+        val baseError = validate(err)
+        baseError.log()
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPackageManagementService.scala:151]: Could this error be one that logs on creation and we could have double logging?

**Review by pbatko-da — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPackageManagementService.scala (hunk 2 above)
Comment (tudor-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiPackageManagementService.scala:151]: `BaseError.Impl` is not logging on creation. Indeed someone could override this when defining the error implementation, but we don't have such implementations so far. The only errors that log on creation atm are the `LoggingTransactionErrorImpl`

File: compatibility/bazel_tools/testing.bzl (hunk 1 above)
Comment (tudor-da) [reply compatibility/bazel_tools/testing.bzl:558]: Updated it

**Review by mziolekda — APPROVED** [review]

File: language-support/hs/bindings/test/DA/Ledger/Tests.hs [hunk 3]
```diff
@@ -356,7 +356,7 @@ tUploadDarFileBad withSandbox = testCase "tUploadDarFileBad" $ run withSandbox $
     lid <- getLedgerIdentity
     let bytes = BS.fromString "not-the-bytes-for-a-darfile"
     Left err <- uploadDarFileGetPid lid bytes
-    liftIO $ assertTextContains err "Invalid DAR: package-upload"
+    liftIO $ assertTextContains err "Dar file is corrupt"
```
Comment (tudor-da) [inline language-support/hs/bindings/test/DA/Ledger/Tests.hs:359]: @cocreature FYI

File: compatibility/bazel_tools/testing.bzl (hunk 1 above)
Comment (tudor-da) [reply compatibility/bazel_tools/testing.bzl:558]: @pbatko-da how I understand it: does not run PackageManagementServiceIT if ledger-api-test-tool is older than 1.18.0-snapshot.20211102.8257.1 (inclusive) and if Ledger API is newer than the same version.

**Pull request merged**